src/core/service.c

```diff
@@ -3181,6 +3144,21 @@ static int service_serialize(Unit *u, FILE *f, FDSet *fds) {
                 (void) serialize_item_format(f, "fd-store-fd", "%i \"%s\" %s", copy, c, one_zero(fs->do_poll));
         }
 
+        FOREACH_ARRAY(i, s->extra_fds, s->n_extra_fds) {
```

Comment:
Thanks for catching this! Sorry I forgot this. Seems this is used when PID1/service manager restarts in order to not lose state (let me know if that's wrong). If so, I can add an integration test for this once your PR merges

Reply:
Yes, please :)

Reply:
@YHNdnzj https://github.com/systemd/systemd/pull/34733

Thanks again!